pkg/tenant/ensign.go

```diff
@@ -0,0 +1,201 @@
+package tenant
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/rotationalio/ensign/pkg/tenant/db"
+	"github.com/rotationalio/ensign/pkg/utils/metatopic"
+	sdk "github.com/rotationalio/go-ensign"
+	api "github.com/rotationalio/go-ensign/api/v1beta1"
+	mimetype "github.com/rotationalio/go-ensign/mimetype/v1beta1"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
+
+// TopicSubscriber is a struct with a go routine that subscribes to the Ensign
+// "meta topic" topic and applies asynchronous updates to the topics in the Tenant
+// database.
+type TopicSubscriber struct {
+	client *EnsignClient
+	stop   chan struct{}
+}
+
+func NewTopicSubscriber(client *EnsignClient) *TopicSubscriber {
+	return &TopicSubscriber{
+		client: client,
+	}
+}
+
+// Run the topic subscriber under the waitgroup.
+func (s *TopicSubscriber) Run(wg *sync.WaitGroup) error {
+	if s.stop != nil {
+		return errors.New("topic subscriber is already running")
+	}
+
+	if wg != nil {
+		wg.Add(1)
+	}
```

Comment:
Why all the niladic checks? If they're necessary a comment about why `wg` could be nil would be good. If they're unnecessary I propose we just make sure we're always passing in a waitgroup or at least an interface that wait group implements.